src/sinks/vector/v2.rs

```diff
@@ -0,0 +1,339 @@
+    address: String,
+    #[serde(default)]
+    pub batch: BatchConfig,
+    #[serde(default)]
+    pub request: TowerRequestConfig,
+    #[serde(default)]
+    pub tls: Option<GrpcTlsConfig>,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct GrpcTlsConfig {
+    ca_file: PathBuf,
+    crt_file: PathBuf,
+    key_file: PathBuf,
+}
+
+impl GenerateConfig for VectorConfig {
+    fn generate_config() -> toml::Value {
+        toml::Value::try_from(default_config("127.0.0.1:6000")).unwrap()
+    }
+}
+
+fn default_config(address: &str) -> VectorConfig {
+    VectorConfig {
+        address: address.to_owned(),
+        batch: BatchConfig::default(),
+        request: TowerRequestConfig::default(),
+        tls: None,
+    }
+}
+
+lazy_static! {
+    static ref REQUEST_DEFAULTS: TowerRequestConfig = TowerRequestConfig {
+        ..Default::default()
+    };
+}
+
+/// grpc doesn't like an address without a scheme, so we default to http if one isn't specified in
+/// the address.
+fn default_http(address: &str) -> crate::Result<Uri> {
+    let uri: Uri = address.parse()?;
+    if uri.scheme().is_none() {
+        // Default the scheme to http.
+        let mut parts = uri.into_parts();
+        parts.scheme = Some("http".parse().expect("http should be valid"));
+        if parts.path_and_query.is_none() {
+            parts.path_and_query = Some("/".parse().expect("root should be valid"));
+        }
+        Ok(Uri::from_parts(parts)?)
+    } else {
+        Ok(uri)
+    }
+}
+
+impl VectorConfig {
+    pub(crate) async fn build(&self, cx: SinkContext) -> crate::Result<(VectorSink, Healthcheck)> {
+        let endpoint = Endpoint::from(default_http(&self.address)?);
+        let endpoint = match &self.tls {
+            Some(tls) => {
+                let host = get_authority(&self.address)?;
+                let ca = Certificate::from_pem(tokio::fs::read(&tls.ca_file).await?);
+                let crt = tokio::fs::read(&tls.crt_file).await?;
+                let key = tokio::fs::read(&tls.key_file).await?;
+                let identity = Identity::from_pem(crt, key);
+
+                let tls_config = ClientTlsConfig::new()
+                    .identity(identity)
+                    .ca_certificate(ca)
+                    .domain_name(host);
+
+                endpoint.tls_config(tls_config)?
+            }
+            None => endpoint,
+        };
+
+        let client = proto::Client::new(endpoint.connect_lazy()?);
+
+        let healthcheck_client = if let Some(uri) = cx.healthcheck.uri.clone() {
+            let endpoint = Endpoint::from(uri.uri);
+            proto::Client::new(endpoint.connect_lazy()?)
+        } else {
+            client.clone()
+        };
+
+        let healthcheck = healthcheck(healthcheck_client, cx.healthcheck.clone());
+
+        let request = self.request.unwrap_with(&REQUEST_DEFAULTS);
+        let batch = BatchSettings::default()
+            .bytes(1300)
```

Comment:
This seems very small for a byte limit, especially considering the event limit is `1000` below.

Reply:
Any ideas what a good limit would be?

It seems every sink has different settings.

For now I've set it to 

```rust
            .bytes(bytesize::mib(10u64))
```

as that is the biggest I can find in use currently.

Reply:
Somewhere between 100x and 1000x the maximum number of events is a reasonable rule of thumb, as that covers a reasonable range of average event size. Usually you would set it based on limits enforced by the underlying service (ie AWS only permits requests less than 1MB or whatever) but that's not the case here. Note that leaving out the `.bytes()` leaves that as effectively unlimited, which might be best.